src/main/java/digital/slovensko/autogram/core/visualization/DocumentVisualizationBuilder.java

```diff
@@ -77,6 +94,79 @@ private Visualization createVisualization(SigningJob job, MimeType transformatio
         return new UnsupportedVisualization(job);
     }
 
+    private DSSDocument getOriginalDocument(DSSDocument document) {
+        SignedDocumentValidator documentValidator = SignedDocumentValidator.fromDocument(document);
+        documentValidator.setCertificateVerifier(new CommonCertificateVerifier());
+        List<AdvancedSignature> signatures = documentValidator.getSignatures();
+        if (signatures.isEmpty()) {
+            throw new RuntimeException("No signatures in document");
+        }
+        AdvancedSignature advancedSignature = signatures.get(0);
+        List<DSSDocument> originalDocuments = documentValidator.getOriginalDocuments(advancedSignature.getId());
+        if (originalDocuments.isEmpty()) {
+            throw new RuntimeException("No original documents found");
+        }
+        return originalDocuments.get(0);
+    }
+
+    private void setMimeTypeFromManifest(DSSDocument originalDocument, DSSDocument document) {
+        DSSDocument manifest = getManifest(originalDocument);
+        if (manifest == null) {
+            return;
+        }
+
+        String documentName = document.getName();
+        MimeType mimeType = getMimeTypeFromManifest(manifest, documentName);
+        if (mimeType == null) {
+            return;
+        }
+
+        document.setMimeType(mimeType);
```

Comment:
a by default to nenastaví podľa manifestu, hej?

Reply:
Ano, iba pri xml